Development

shandralor opened the pull request:
Added remove user friends functionality. Not sure what needs to happen next, but I think you need to merge some conflicts @Lukas-Batema

luni-moon committed:
Update gitignore, update login, update req file

luni-moon committed:
Merge branch 'shandralor:main' into lukas-batema/development

luni-moon committed:
Working on return types

Commit:
Keep working on type mismatching errors

luni-moon committed:
Still working on the base64 None str

luni-moon committed:
Complete the error handling for the password

Commit:
Merge branch 'main' into lukas-batema/development

shandralor committed:
fixed login token bug

luni-moon committed:
Merge branch 'lukas-batema/development' into development

shandralor committed:
Added "remove a friend" + start code DM functions

shandralor committed:
Merge branch 'development' of github.com:shandralor/PeerBrain into development

shandralor committed:
Merge branch 'main' into development

shandralor committed:
Added option 7 in the menu to test message viewing

shandralor committed:
Merge branch 'development' of github.com:shandralor/PeerBrain into development

shandralor commented:
Please review the option 7 push also. It implements viewing message titles only and being able to choose an individual message to view. If you view it, the rating increase function will get triggered.

shandralor committed:
Added html page to email verification.

Lukas-Batema commented on server/db.py:
```diff
@@ -334,3 +375,72 @@ def get_encrypted_sym_key(username: str, user_password, friend_username:str):
+            logging.exception(error_message)
+            return
+
+update_thought_rating("222a8c6f-13a0-4f4c-8ae9-8f1e2dae6b8a")
```
What is this? I'd put this in an environment variable if I were you.

shandralor replied:
it is a dynamically generated hash based on the two usersnames. It will be used to easily retrieve conversations. If you know the username and the friends username the correct hash will be generated(they are sorted alphabetically and trimmed to 5 characters if longer than 5, before hash generation). It is not so much a security feature. It is more to easily store and retrieve the conversations. There will be one such hash per conversation between two people

shandralor replied:
plus that update_thought_rating at the bottom needs to go, i was testing functionality :-P

Lukas-Batema replied:
Understood

luni-moon reviewed (commented):
I am gonna test this locally, just to be sure, as this is a big change ;)

Lukas-Batema commented on client/cli_main.py:
```diff
@@ -230,6 +266,72 @@ def main():
+                        remove_user_friends(server_url, friend_username)
+                        #reloading friends object after removing a friend
+                        friends = get_user_friends(server_url)
+                    elif sub_choice == "7":
```
Currently this is giving me this (error at the bottom):
```txt
PS D:\Programming\tmp\PeerBrain\client> python cli_main.py

Welcome to version 0.2 of the Brainwaves P2P client!
-------------------------------------------------------------------
The currently set server url is https://peerbrain.teckhawk.be/
-------------------------------------------------------------------

Please select what you want to do from the menu below.
1. Log in to the current server
2. Register account on the current server
3. Change server
4. Reset your password
Q to exit
>> 1

Logged in with valid token


MAIN MENU:


Please choose an option:

1. TECHNICAL ACTIONS
2. ACCOUNT ACTIONS
3. LOG OUT
Q to exit
>> 2

ACCOUNT MENU:


Please choose an option:

1. Check your account details
--------------------------------
2. Create a message
3. Show all messages from a friend
--------------------------------
4. Add a friend
5. Check friends list
6. Remove a friend from your friend list
--------------------------------
7. New thought display function test
B to return to main menu
>> 7

Please confirm your password to get your messages:


Please enter the username of the friend that you want to see messages from:

admin
Please choose a thought to read by entering its number or type B to go back:

1. TITLE: testing as admin       RATING: 1.2375035237499352
2. TITLE: testing message for nishant    RATING: 1.2375035237499352
3. TITLE: testing message while refactor cli     RATING: 2.1724833783268154
4. TITLE: The meaning of life?   RATING: 1.2375035237499352

Enter thought number: 4

TITLE:  The meaning of life?
RATING: 1.2375035237499352

MESSAGE:  You thought there would be something here, didn't you :-P.


Traceback (most recent call last):
  File "D:\Programming\tmp\PeerBrain\client\cli_main.py", line 360, in <module>
    main()
  File "D:\Programming\tmp\PeerBrain\client\cli_main.py", line 325, in main
    update_rating_for_thought(server_url, str(selected_thought["key"]))
KeyError: 'key'
NativeCommandExitException: Program "python.exe" ended with non-zero exit code: 1.
```

shandralor replied:
It is already fixed in the server code. The get thoughts function did not return the key initially. I've made changes that do return it, but you need to run the dev server. You are testing this on main, which does not have the altered server code yet.

Lukas-Batema replied:
My bad, will test on dev

luni-moon reviewed (changes requested):
Currently, I'm getting this error on option 7, will fix tonight!

Commit:
Sort of fix it? It still doesn't update the rates

Lukas-Batema commented on client/cli_main.py:
```diff
@@ -306,35 +307,38 @@ def main():
+                                            key_path = os.path.join(os.path.dirname(__file__), 'keys', 'message.key')
+                                            with open(key_path, 'rb') as key_file:
+                                                key = key_file.readline(1)
+                                            update_rating_for_thought(server_url, key)
```
I'm not sure the key is supposed to be a str type, so I changed it to bytes, but it still is throwing a 404 error...

shandralor replied:
it is a string type normally, see my previous answer :-)

Lukas-Batema commented on client/client_functions.py:
```diff
@@ -176,19 +176,19 @@ def get_user_friends(server_url:str)->tuple:
+    params = {"key": key}
     
-    response = requests.get(f"{server_url}{account_url_suffix}", headers=headers, params = params, timeout=10)
+    response = requests.get(f"{server_url}{account_url_suffix}", headers=headers, params=params["key"], timeout=10)
```
Had to update this line in accordance to the return type screams

shandralor replied:
that is kinda ridiculous lol... I wasn't even aware this was a thing in python

luni-moon reviewed (changes requested):
Still needs to be fixed, so I'm requesting changes

shandralor committed:
Direct Messaging functional, option 8

shandralor committed:
Merge branch 'development' of github.com:shandralor/PeerBrain into development

Commit:
mid-work, NOT READY

Commit:
Should work now.

Lukas-Batema commented on client/cli_main.py:
```diff
@@ -163,7 +188,10 @@ def main():
                     print("5. Check friends list")
+                    print("6. Remove a friend from your friend list")
                     print("--------------------------------")
+                    print("7. New thought display function test")
```
You may want to change this name to something better now.

Lukas-Batema commented on client/cli_main.py:
```diff
@@ -228,11 +268,91 @@ def main():
+                                        print("You selected a non-existant thought number.\nError:", err)
+                    elif sub_choice == "8":
+                        friend_username = input("Please enter your friends username: \n")
+                        get_user_conversation(server_url, friend_username)
```
I'm getting this error (whole output, error near bottom (404 doubled):
```txt
@Lukas-Batema ➜ /workspaces/PeerBrain (development) $ python client/cli_main.py

Welcome to version 0.2 of the Brainwaves P2P client!
-------------------------------------------------------------------
The currently set server url is https://peerbrain.teckhawk.be/
-------------------------------------------------------------------

Please select what you want to do from the menu below.
1. Log in to the current server
2. Register account on the current server
3. Change server
4. Reset your password
Q to exit
>> 1

Logged in with valid token


MAIN MENU:


Please choose an option:

1. TECHNICAL ACTIONS
2. ACCOUNT ACTIONS
3. LOG OUT
Q to exit
>> 2

ACCOUNT MENU:


Please choose an option:

1. Check your account details
--------------------------------
2. Create a message
3. Show all messages from a friend
--------------------------------
4. Add a friend
5. Check friends list
6. Remove a friend from your friend list
--------------------------------
7. New thought display function test
8. Private conversations(DM)
B to return to main menu
>> 8
Please enter your friends username: 
admin



--------
{'detail': 'Not Found'}
--------
{'detail': 'Not Found'}
Error: 404

1. Yes
2. No
Do you want to reply to admin? 1
Enter your message: joe
Something went wrong with sending your message!


ACCOUNT MENU:


Please choose an option:

1. Check your account details
--------------------------------
2. Create a message
3. Show all messages from a friend
--------------------------------
4. Add a friend
5. Check friends list
6. Remove a friend from your friend list
--------------------------------
7. New thought display function test
8. Private conversations(DM)
B to return to main menu
>> b
Returning to main menu...

MAIN MENU:


Please choose an option:

1. TECHNICAL ACTIONS
2. ACCOUNT ACTIONS
3. LOG OUT
Q to exit
>> q
Goodbye!
```
Will that be fixed with the dev database that I don't have access to?

Lukas-Batema replied:
Okay, it should just need these changes (basically renaming the option 7 of main option 2)

shandralor replied:
done :-)

luni-moon reviewed (changes requested):
Almost good. These still need to be changed.

shandralor committed:
Name change option 7

shandralor committed:
Merge branch 'development' of github.com:shandralor/PeerBrain into development

Lukas-Batema commented on client/cli_main.py:
```diff
@@ -163,7 +188,10 @@ def main():
                     print("5. Check friends list")
+                    print("6. Remove a friend from your friend list")
                     print("--------------------------------")
+                    print("7. New system to view messages individually. This will also increase their rating.")
```
I'd reword it like so, "View your messages singularly".

luni-moon reviewed (approved):
Looks good!

**EDIT**: Please refer to the new review...

luni-moon reviewed (changes requested):
Meant to request the last review as a change, wasn't thinking haha.

shandralor committed:
changed cli menu wording

shandralor commented:
done

shandralor committed:
Db code refactor for users and pw reset done

shandralor committed:
modules clean up

shandralor committed:
bug fix

shandralor committed:
API Endpoint update to use refactored func

shandralor committed:
db refactor almost complete, do not merge

shandralor committed:
DB CODE REFACTOR TO MONGO COMPLETE

Commit:
Let's try this again, shall we?

shandralor committed:
Merge pull request #21 from shandralor/development_save

Let's try this again, shall we?

Commit:
Add the GUI Client

Commit:
Add Gui

shandralor committed:
Merge pull request #20 from shandralor/db-refactor

Fix db-refactor branch

Commit:
View Thoughts!

Commit:
Update friend.html

Commit:
Create enterpass.html

Commit:
Rename cli_main.py to __main__.py

Commit:
Rename main.py to __main__.py

Commit:
Rename main.py to __main__.py

Commit:
Rename main.py to __main__.py

shandralor committed:
Merge pull request #26 from WilliamDavidHarrison/patch-1

Use __main__.py

Commit:
ADD 2FA support

Commit:
ADD 2FA support

Commit:
Add otp_secret to the model

otp_secret (str, optional): The user's one-time password secret. Defaults to None.

Commit:
Final

shandralor committed:
Merge pull request #27 from shandralor/andrewstech-patch-1

ADD 2FA Support

Commit:
Add Bug reporting

shandralor committed:
Server bugfixes

shandralor committed:
Merge branch 'development' of github.com:shandralor/PeerBrain into development

Commit:
Bug fixes

Commit:
FIX USERNAME FOR EMAIL

Commit:
Add missing :

Commit:
FIx dupe messages
